bike_kollective/lib/data/provider/authentication.dart

```diff
@@ -13,30 +17,90 @@ final authenticationProvider = Provider<Authentication>((ref) {
 
 // Interface for doing authentication
 abstract class Authentication {
-  Future<Uid> signInWithGoogle();
+  Future<AuthResult?> signInWithGoogle();
 }
 
 // This implementation can be used by developers to create fake data
 // Note: be sure to return DummyAuthentication in the authenticationProvider above
 class DummyAuthentication extends Authentication {
   @override
-  Future<Uid> signInWithGoogle() async {
-    return "DUMMY_GOOGLE_USER";
+  Future<AuthResult> signInWithGoogle() async {
+    return AuthResult(uid: "Dummer_UID", email: "Dummy_Email");
   }
 }
 
 // This implementation requests
 class RealAuthentication extends Authentication {
   @override
-  Future<Uid> signInWithGoogle() async {
+  Future<AuthResult> signInWithGoogle() async {
     GoogleSignInAccount? googleUser = await GoogleSignIn().signIn();
     GoogleSignInAuthentication? googleAuth = await googleUser?.authentication;
     AuthCredential credentials = GoogleAuthProvider.credential(
       accessToken: googleAuth?.accessToken,
       idToken: googleAuth?.idToken,
     );
-    UserCredential userCredential =
-        await FirebaseAuth.instance.signInWithCredential(credentials);
-    return userCredential.user!.uid;
+
+    UserCredential userCredential = 
+      await FirebaseAuth.instance.signInWithCredential(credentials);
+
+    String uid = userCredential.user!.uid;
+    String? email = userCredential.user?.email;
+    print(userCredential.user!.uid);
+    print(userCredential.user!.email);
+
+    return AuthResult(
+      uid: uid, 
+      email: email,
+    );
   }
 }
+
+class AuthResult {
+  final String uid;
+  final String? email;
+
+  AuthResult({required this.uid, this.email});
+}
+
+// Provides access to login with email
+final emailAuthProvider = Provider<EmailAuthentication>((ref) {
```

Comment:
Combine with existing Authentication class and clean up the code. There's no need for these to be separate. Also, rename "log in" to "sign in" to be consistent.